### ISSUE TITLE
All 30B and 65B ggml models now generate nonsense

### ISSUE THREAD

**Issue by vmajor** [abetlen/llama-cpp-python#475]
# Expected Behavior

Sane output as with v0.16.9

# Current Behavior

Please provide a detailed written description of what `llama-cpp-python` did, instead.

"### Human Rights Reserved."

"-\n\\end{code}\n\nComment: I'm not sure what you are asking here. Can you please rephrase your question? What is it that you want to do with this string? Do you just want to remove all non-printable characters from the string or something else?\n\nAnswer: You can use `string.replace` method like below :\n\n\\begin{code}\nstr = \"Hello World! \\n How are you?\"\nnew_str = str.replace(\"\\n\", \"\") # new_str will be \"Hello World! How are you?\"\n\\end{code}"

 "s-18000-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-1800-18"

"\u200d\u200bQQQQQQQQNNNNNMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN"

# Environment and Context

v0.1.70

```
$ python3 3.9.10

```

# Please provide detailed steps for reproducing the issue. We are not sitting in front of your screen, so the more detail the better.

Run any inference code from 0.16.9 with any 65B model (quantized or not) and observe the output.

**Comment by vmajor** [comment]
Problem persists with 0.17.1

Prompt: """ Read this carefully, and reflect on your answer before you give it: David has three sisters. How many brothers does each sister have?"""

Answer: " automatisch in der Datenbank von WikidataVorlage:Infobox-Karte/Wartung/Magnitude fehlt

Die folgende Tabelle zeigt die größten Erdbeben seit dem Jahr 1900 mit einer Magnitude ab 7,5 auf der Momenten-Magnitudenskala und darüber hinaus. Die Angaben zu den Opferzahlen beruhen meist auf Schätzungen oder Hochrechnungen; genauere Zahlen sind nicht immer verfügbar.

## Siehe auch

* Liste historischer Erdbeben
* Liste von Katastrophen"

**Comment by vmajor** [comment]
Downgrading to 0.1.69 does not help

**Comment by gjmulder** [comment]
I'm busily requantizing my 65B model... patience pls :smile:

**Comment by gjmulder** [comment]
```
$ pip list | grep llama
llama-cpp-python         0.1.71

$ grep llama_cpp smoke_test.py 
from llama_cpp import Llama

$ python ./smoke_test.py -f ./prompt.txt -m /data/llama/65B/ggml-model-q5_1.bin --n_ctx 512 --max_tokens 256
llama.cpp: loading model from /data/llama/65B/ggml-model-q5_1.bin
llama_model_load_internal: format     = ggjt v3 (latest)
llama_model_load_internal: n_vocab    = 32000
llama_model_load_internal: n_ctx      = 512
llama_model_load_internal: n_embd     = 8192
llama_model_load_internal: n_mult     = 256
llama_model_load_internal: n_head     = 64
llama_model_load_internal: n_layer    = 80
llama_model_load_internal: n_rot      = 128
llama_model_load_internal: ftype      = 9 (mostly Q5_1)
llama_model_load_internal: n_ff       = 22016
llama_model_load_internal: model size = 65B
llama_model_load_internal: ggml ctx size =    0.19 MB
llama_model_load_internal: mem required  = 50301.80 MB (+ 5120.00 MB per state)
llama_new_context_with_model: kv self size  = 1280.00 MB
AVX = 1 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | FMA = 1 | NEON = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 0 | SSE3 = 1 | VSX = 0 | 

llama_print_timings:        load time = 21504.86 ms
llama_print_timings:      sample time =   188.57 ms /   256 runs   (    0.74 ms per token,  1357.57 tokens per second)
llama_print_timings: prompt eval time = 21504.81 ms /    30 tokens (  716.83 ms per token,     1.40 tokens per second)
llama_print_timings:        eval time = 211355.20 ms /   255 runs   (  828.84 ms per token,     1.21 tokens per second)
llama_print_timings:       total time = 233529.47 ms
{
  "id": "cmpl-ea6e79be-615a-422b-b684-817d40ee0756",
  "object": "text_completion",
  "created": 1689328222,
  "model": "/data/llama/65B/ggml-model-q5_1.bin",
  "choices": [
    {
      "text": "Read this carefully, and reflect on your answer before you give it: David has three sisters. How many brothers does each sister have?\nMost people will say \"three\" because that's the immediate answer they see in their mind's eye. But actually, the correct answer is two\u2014one fewer than you thought at first glance. Here's why: You need to stop and analyze what David means when he says \"sister.\"\nWhen I was in high school, my brother started dating a girl named Shirley who became his main squeeze for many years. One day my mother asked me if I knew about Shirley's sister. She said that a mutual friend had told her about the sister and that she would be bringing her to our house one night to meet us, because she was interested in going out with my brother on a date.\n\"Shirley has a sister?\" I asked incredulously. \"Since when does Shirley have a sister? Who is it?\"\nMom looked at me oddly and said, \"Well, you know, the girl she's related to.\"\n\"Related to?\" I replied. \"You mean like her mom or something? Like a cousin?\"\nMother shook her head and smiled, but not because she was amused. She was just used to it. \"",
      "index": 0,
      "logprobs": null,
      "finish_reason": "length"
    }
  ],
  "usage": {
    "prompt_tokens": 30,
    "completion_tokens": 256,
    "total_tokens": 286
  }
}

```

**Comment by vmajor** [comment]
Lol :) If it helps it also fails with the 30B Tulu and 30B based.

**Comment by gjmulder** [comment]
65B is working fine for me...

**Comment by vmajor** [comment]
...ohforf... did you requantize it?

**Comment by vmajor** [comment]
I even created  new venv and I still get nonsense responses. I then tested llama.cpp directly just to make sure that my environment is building it correctly and I have no errors.

So, something really strange is happening with my llama-cpp-python and I cannot fix it by downgrading, or uninstalling and reinstalling, or even by creating a new venv for it. Any advice on how I can diagnose what is happening?

Everything was working well until the 11th of July.

**Comment by gjmulder** [comment]
I used the latest `llama.cpp` (re)quantization as I deleted all the quantized 30B and 65B llama models due to the disk space reqs, but this doesn't look to be your problem

1. I'd run `pip uninstall llama-cpp-python` multiple times to ensure you don't have another instance in your `PYTHONPATH`
2. Try building `llama-cpp-python` from a newly git cloned source w/recurse submodules to pull in the latest `llama.cpp`
3. Before source build run `rm -rf _skbuild/` to nuke any old llama.cpp shared libs and broken compiles (but if you did a new `git clone...` you shouldn't of course have to).

**Comment by vmajor** [comment]
OK thank you for that. When I build from source and running the recursive fetch the package builds but I get this error at runtime:
```
Traceback (most recent call last):
  File "/home/vmajor/advisor/llm_test.py", line 3, in <module>
    from llama_cpp import Llama
  File "/home/vmajor/llama-cpp-python/llama_cpp/__init__.py", line 1, in <module>
    from .llama_cpp import *
  File "/home/vmajor/llama-cpp-python/llama_cpp/llama_cpp.py", line 334, in <module>
    _lib.llama_backend_init.argtypes = [c_bool]
  File "/home/vmajor/anaconda3/envs/advisor/lib/python3.10/ctypes/__init__.py", line 387, in __getattr__
    func = self.__getitem__(name)
  File "/home/vmajor/anaconda3/envs/advisor/lib/python3.10/ctypes/__init__.py", line 392, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))
AttributeError: /home/vmajor/llama-cpp-python/llama_cpp/libllama.so: undefined symbol: llama_backend_init
```

**Comment by gjmulder** [comment]
Yeah, I got that bug too, so I did a package install:

`pip install llama-cpp-python --force-reinstall --upgrade --no-cache-dir`

**Comment by vmajor** [comment]
...sigh... still does not work. This is so frustrating. Do you know where llama-cpp-python is storing its version of llama.cpp? I am thinking of linking the directory to my llama.cpp directory that contains working code and see if that is what's causing this ridiculous issue that I am struggling with.

**Comment by gjmulder** [comment]
> ...sigh... still does not work. This is so frustrating. Do you know where llama-cpp-python is storing its version of llama.cpp? I am thinking of linking the directory to my llama.cpp directory that contains working code and see if that is what's causing this ridiculous issue that I am struggling with.

That's why I suggested building from source as then you can diagnose the compile and look what it has and hasn't built. Something looks broken in your environment, possibly at the OS level if a new venv didn't work. Or maybe try python v3.9 which will mix up the python dep versions a bit?

Finally, you can use git to revert to a commit that doesn't throw #479, which is likely a `llama.cpp` induced bug.

**Comment by gjmulder** [comment]
> ...ohforf... did you requantize it?

Welcome to Python. Sanity is optional. A sense of humour is required.

![image](https://github.com/abetlen/llama-cpp-python/assets/10833500/343529e7-3226-4a9c-86ba-abe1294fc61b)

If it is any consolation, Windoze is worse. I simply won't touch CUDA on Windoze.

This is why I use Docker images so I can tightly control OS, OS package, Python and Python package versioning. I always keep a few previous Docker images around to ensure I have stable environments. I only update when forced to, and let other people find the bugs :wink: :wine_glass:

**Comment by vmajor** [comment]
well I searched the file system for `libllama.so` and there were no insidious holdouts sabotaging my work. I did however activate some of my less active venvs and ran my test code with llama-cpp-python 0.1.57 and it worked fine.

So whatever insanity is possessing my hitherto primary working environment is in fact related to the latest versions of something that llama-cpp-python is installing. They just do not run in a sane way on my hardware.

`lscpu`
```
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         48 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  24
  On-line CPU(s) list:   0-23
Vendor ID:               AuthenticAMD
  Model name:            AMD Ryzen 9 3900XT 12-Core Processor
    CPU family:          23
    Model:               113
    Thread(s) per core:  2
    Core(s) per socket:  12
    Socket(s):           1
    Stepping:            0
    BogoMIPS:            8400.05
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constan
                         t_tsc rep_good nopl tsc_reliable nonstop_tsc cpuid extd_apicid pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervis
                         or lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext perfctr_core ssbd ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 r
                         dseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves clzero xsaveerptr arat npt nrip_save tsc_scale vmcb_clean flushbyasid decodeassists
                         pausefilter pfthreshold v_vmsave_vmload umip rdpid
Virtualization features:
  Virtualization:        AMD-V
  Hypervisor vendor:     Microsoft
  Virtualization type:   full
Caches (sum of all):
  L1d:                   384 KiB (12 instances)
  L1i:                   384 KiB (12 instances)
  L2:                    6 MiB (12 instances)
  L3:                    16 MiB (1 instance)
NUMA:
  NUMA node(s):          1
  NUMA node0 CPU(s):     0-23
Vulnerabilities:
  Itlb multihit:         Not affected
  L1tf:                  Not affected
  Mds:                   Not affected
  Meltdown:              Not affected
  Mmio stale data:       Not affected
  Retbleed:              Mitigation; untrained return thunk; SMT enabled with STIBP protection
  Spec store bypass:     Mitigation; Speculative Store Bypass disabled via prctl and seccomp
  Spectre v1:            Mitigation; usercopy/swapgs barriers and __user pointer sanitization
  Spectre v2:            Mitigation; Retpolines, IBPB conditional, STIBP always-on, RSB filling, PBRSB-eIBRS Not affected
  Srbds:                 Not affected
  Tsx async abort:       Not affected
```
```
uname -a
Linux DESKTOP-1TO72R9 5.15.68.1-microsoft-standard-WSL2+ #2 SMP Sun Oct 2 09:50:15 CEST 2022 x86_64 x86_64 x86_64 GNU/Linux
```

```
python --version
Python 3.10.9
```

**Comment by vmajor** [comment]
OK I fixed the problem, for me, as it seems that there is no problem with other systems.

1. Clone llama-cpp-python repo, but **do not** allow it to fetch the llama.cpp branch that it wants
```
git clone git@github.com:abetlen/llama-cpp-python.git

```
2. Remove the empty `llama.cpp` and clone the current master `llama.cpp` inside `vendor `instead
```
cd ~/llama-cpp-python/vendor/
rm -rf llama.cpp
git clone https://github.com/ggerganov/llama.cpp.git
```
3. Build the packages and  create libllama.so
`python setup.py develop`

4. Copy the resulting libllama.so into your venv
`cp ~/llama-cpp-python/vendor/llama.cpp/libllama.so ~/anaconda3/envs/[your_venv_name]/lib/[your_python_version]/site-packages/llama_cpp/`

5. Enjoy the fully working code, again

I do not know enough about what happened with the llama.cpp version that llama-cpp-python pulls and why it specifically dislikes my system, but it is a fix for this situation. I did notice some perhaps related bug report in llama.cpp but it was centered around nonsensical responses when using CUBLAS, not CPU.
